aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java

```diff
@@ -0,0 +1,260 @@
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(DeleteTableResponse.builder().build())
+        .when(glue).deleteTable(Mockito.any(DeleteTableRequest.class));
+    glueCatalog.dropTable(TableIdentifier.of("db1", "t1"));
+  }
+
+  @Test
+  public void renameTable() {
+    Map<String, String> properties = new HashMap<>();
+    properties.put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE);
+    Mockito.doReturn(GetTableResponse.builder()
+        .table(Table.builder().databaseName("db1").name("t1").parameters(properties).build()).build())
+        .when(glue).getTable(Mockito.any(GetTableRequest.class));
+    Mockito.doReturn(GetTablesResponse.builder().build())
+        .when(glue).getTables(Mockito.any(GetTablesRequest.class));
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(DeleteTableResponse.builder().build())
+        .when(glue).deleteTable(Mockito.any(DeleteTableRequest.class));
+    Mockito.doReturn(CreateTableResponse.builder().build())
+        .when(glue).createTable(Mockito.any(CreateTableRequest.class));
+    glueCatalog.dropTable(TableIdentifier.of("db1", "t1"));
+  }
+
+  @Test
+  public void createNamespace() {
+    Mockito.doReturn(CreateDatabaseResponse.builder().build())
+        .when(glue).createDatabase(Mockito.any(CreateDatabaseRequest.class));
+    glueCatalog.createNamespace(Namespace.of("db"));
+  }
+
+  @Test(expected = NoSuchNamespaceException.class)
+  public void createNamespace_badName() {
+    Mockito.doReturn(CreateDatabaseResponse.builder().build())
+        .when(glue).createDatabase(Mockito.any(CreateDatabaseRequest.class));
+    glueCatalog.createNamespace(Namespace.of("db-1"));
+  }
+
+  @Test
+  public void listNamespaces_all() {
+    Mockito.doReturn(GetDatabasesResponse.builder()
+        .databaseList(
+            Database.builder().name("db1").build(),
+            Database.builder().name("db2").build()
+        ).build())
+        .when(glue).getDatabases(Mockito.any(GetDatabasesRequest.class));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            Namespace.of("db1"),
+            Namespace.of("db2")
+        ),
+        glueCatalog.listNamespaces()
+    );
+  }
+
+  @Test
+  public void listNamespaces_self() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            Namespace.of("db1")
+        ),
+        glueCatalog.listNamespaces(Namespace.of("db1"))
+    );
+  }
+
+  @Test(expected = NoSuchNamespaceException.class)
```

Comment:
It is usually better to use `AssertHelpers.assertThrows`. That way, you can check the error message and run additional assertions after the exception. For example, if you're testing create when a table already exists, you can validate that the right exception is thrown and that the existing table was not modified.